### DISCUSSION TITLE
How to enable PCRE2 when cargo installing rg?

### DISCUSSION THREAD

**Discussion by jugmac00** [BurntSushi/ripgrep#2190] [Q&A] [answered]
AFAIR I did

```
cargo install ripgrep
```

When I try to use rg with pcre...

```
rg --pcre2 "(\b\S+\b)\s+\b\1\b"
PCRE2 is not available in this build of ripgrep
```

When googling the error message I found your faq entry
https://github.com/BurntSushi/ripgrep/blob/master/FAQ.md#how-do-i-use-lookaround-andor-backreferences

> If you installed ripgrep through a different means (like your system's package manager), then please reach out to the maintainer of that package to see whether it's possible to enable the PCRE2 feature.

What would I need to do to get pcre support?

Thank you!

**Comment by BurntSushi** [comment, marked as answer]
`cargo install ripgrep` is just a shortcut for building the latest release of ripgrep from source. So the [build instructions](https://github.com/BurntSushi/ripgrep#building) apply. Instead of running `cargo build --release --features 'pcre2'`, you'd run `cargo install --features 'pcre2'`. (You don't need the `--release` flag since the `install` sub-command enables that automatically.)

**Reply by azzamsa** [reply]
For the newer cargo version, it is `cargo install --path . --features 'pcre2'`

**Reply by ltrzesniewski** [reply]
@azzamsa There's a semantic difference though:
- `cargo install ripgrep --features pcre2` will install the latest released crate version
- `cargo install --path . --features pcre2` will install from the source code in the current directory

**Reply by azzamsa** [reply]
Wow, I have no clue I can enable cargo features without cloning the project. TIL.
Thanks!
